phantoms/MR_XCAT_qMRI/sim_ivim_sig.py

```diff
@@ -0,0 +1,406 @@
+    Tissue[17] = [576, 46, 812, 42]
+    Tissue[18] = [576, 46, 812, 42]
+    Tissue[19] = [1045.5, 37.3, 1201, 44]
+    Tissue[20] = [981.5, 36, 1232.9, 37.20]
+    #Tissue[20] = [981.5, 36, 1232.9, 37.20]
+    Tissue[21] = [0, 0, 0, 0]
+    Tissue[22] = [584, 46, 725, 43]
+    Tissue[23] = [828, 71, 1168, 66]
+    Tissue[24] = [1412, 85, 1545, 81]
+    Tissue[25] = [828, 71, 1168, 66]
+    Tissue[26] = [1412, 85, 1545, 81]
+    Tissue[27] = [576, 46, 812, 42]
+    Tissue[28] = [200, 0.5, 302, 0.25]
+    Tissue[29] = [200, 0.5, 302, 0.25]
+    Tissue[30] = [1057, 79, 1328, 61]
+    Tissue[31] = [200, 0.5, 302, 0.25]
+    Tissue[32] = [200, 0.5, 302, 0.25]
+    Tissue[33, :] = [200, 0.5, 302, 0.25]
+    Tissue[34, :] = [745, 74, 993, 78]
+    Tissue[35, :] = [549, 49, 586, 49]
+    Tissue[36, :] = [1585, 254, 1664, 147]
+    Tissue[37, :] = [1582, 181, 1584, 66]
+    Tissue[38, :] = [576, 46, 812, 42]
+    Tissue[39, :] = [1317, 88, 1597, 74]
+    Tissue[40, :] = [576, 46, 812, 42]
+    Tissue[41, :] = [576, 46, 812, 42]
+    Tissue[42, :] = [576, 46, 812, 42]
+    Tissue[43, :] = [576, 46, 812, 42]
+    Tissue[44, :] = [576, 46, 812, 42]
+    Tissue[45, :] = [1317, 88, 1597, 74]
+    Tissue[46, :] = [576, 46, 812, 42]
+    Tissue[47, :] = [576, 46, 812, 42]
+    Tissue[48, :] = [576, 46, 812, 42]
+    Tissue[49, :] = [576, 46, 812, 42]
+    Tissue[50, :] = [981.5, 36, 1232.9, 37.20]
+    Tissue[51, :] = [1024, 30, 1168, 27]
+    # Tissue[52, :] = [;;;]
+    Tissue[54, :] = [1434.5, 164, 1498.3, 164]
+    Tissue[55, :] = [1434.5, 164, 1498.3, 164]
+    Tissue[56, :] = [5053, 468, 5053, 468]
+    # Tissue[57, :] = [;;;]
+    Tissue[58, :] = [1045.5, 37.3, 1201, 44]
+    Tissue[59, :] = [0, 0, 0, 0]
+    Tissue[60, :] = [1309, 117, 1514, 79]
+    Tissue[61, :] = [1135, 58, 1616, 83]
+    Tissue[62, :] = [576, 46, 812, 42]
+    Tissue[63, :] = [576, 46, 812, 42]
+    Tissue[64, :] = [576, 46, 812, 42]
+    Tissue[65, :] = [1317, 88, 1597, 74]
+    Tissue[66, :] = [1317, 88, 1597, 74]
+    Tissue[67, :] = [1317, 88, 1597, 74]
+    Tissue[68, :] = [1317, 88, 1597, 74]
+    Tissue[69, :] = [1317, 88, 1597, 74]
+    Tissue[69, :] = [5053, 468, 5053, 468]
+    Tissue[70, :] = [1138, 26, 1138, 26]
+    # Tissue[71, :] = [;;;]
+    # Tissue[72, :] = [312.4, 117.4, 377, 97.5]
+    Tissue[72, :] = [9999999999, 0.00000001, 999999999, 0.00000001] ## fat suppression
+    Tissue[73, :] = [0, 0, 1676, 64]
+    Dim= np.zeros(XCAT.shape, dtype=np.float32)
+    fim= np.zeros(XCAT.shape, dtype=np.float32)
+    Dpim= np.zeros(XCAT.shape, dtype=np.float32)
+    MR = np.zeros(XCAT.shape+(len(bvalue),), dtype=np.float32)
+    for iTissue in range(len(Tissue)):
+        if iTissue != 0:
+            if b0 == 1.5:
+                T1 = Tissue[iTissue, 0]
+                T2 = Tissue[iTissue, 1]
+            else:
+                T1 = Tissue[iTissue, 2]
+                T2 = Tissue[iTissue, 3]
+
+            if ivim_cont and iTissue in tissue_included:
```

Comment:
I'm just noticing this now. If it's IVIM and the tissue is in the included list. Maybe just check if D, f, or Ds are 0 rather than maintaining this other list? Or initialize as NaN and check if they're not NaN?

Reply:
Or I can look at it, but that may be a while.

Reply:
I fixed it in my PR